winch/codegen/src/isa/x64/masm.rs

```diff
@@ -337,14 +339,52 @@ impl Masm for MacroAssembler {
         self.asm.cmp(src.into(), dst, size);
         self.asm.setcc(kind, dst);
     }
+
+    fn clz(&mut self, src: Reg, dst: Reg, size: OperandSize) {
+        if self.flags.has_lzcnt() {
+            self.asm.lzcnt(src, dst, size);
+        } else {
+            let scratch = regs::scratch();
```

Comment:
Is it safe for me to use the `scratch` register like this or do I need to explicitly allocate it first?

Reply:
Your usage here is safe as far as I can tell.